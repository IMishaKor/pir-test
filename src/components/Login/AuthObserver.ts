import { useEffect } from 'react';
import { useStore } from '../../store/AuthStore';

function AuthObserver() {
  const authStore = useStore();

  useEffect(() => {
    const authObserver = (e: StorageEvent): void => {
      if (e.key === 'AUTH_USER_ID') {
        const userId = localStorage.getItem(e.key) || 0;
        authStore?.authMe(+userId);
      }
    };
    window.addEventListener('storage', authObserver, false);

    return () => {
      window.removeEventListener('storage', authObserver, false);
    };
  }, []);
  return null;
}
export default AuthObserver;
