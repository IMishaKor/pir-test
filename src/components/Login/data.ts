export const authFormData = {
  Email: '',
  Password: '',
};
