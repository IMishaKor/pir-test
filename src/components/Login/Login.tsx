import { useState } from 'react';
import { Redirect, useHistory } from 'react-router-dom';
import { observer } from 'mobx-react-lite';
import Box, { Item } from 'devextreme-react/box';
import Form, { ButtonItem, GroupItem, SimpleItem, Label, EmailRule, RequiredRule } from 'devextreme-react/form';
import notify from 'devextreme/ui/notify';
import { useStore } from '../../store/AuthStore';

import { authFormData } from './data';

const Login = () => {
  const authStore = useStore();

  const [formFetch, setFormFetch] = useState(false);
  const history = useHistory();
  const handleSubmit = (e: React.SyntheticEvent): void => {
    e.preventDefault();
    setFormFetch(true);
    authStore
      ?.authLogin(authFormData.Email, authFormData.Password)
      .then((data) => {
        if (data.error) {
          notify(data.message, 'error', 3000);
        } else {
          authFormData.Email = '';
          authFormData.Password = '';
          history.push('/notes');
        }
        setFormFetch(false);
      })
      .catch((error) => {
        notify(error, 'error', 3000);
        setFormFetch(false);
      });
  };

  if (authStore?.isAuth) return <Redirect to="/" />;
  return (
    <Box direction="row" width="100%" align="center" crossAlign="center">
      <Item ratio={0} baseSize={400}>
        <form action="your-action" onSubmit={handleSubmit}>
          <Form
            formData={authFormData}
            labelLocation="top"
            readOnly={false}
            showColonAfterLabel={true}
            validationGroup="customerData"
          >
            <GroupItem caption="Авторизация">
              <SimpleItem dataField="Email" editorType="dxTextBox">
                <Label text="Адрес электронной почты" />
                <RequiredRule message="Email обязателен для заполнения" />
                <EmailRule message="Email неверный" />
              </SimpleItem>
              <SimpleItem dataField="Password" editorType="dxTextBox" editorOptions={{ mode: 'password' }}>
                <Label text="Пароль" />
                <RequiredRule message="Введите пароль" />
              </SimpleItem>
            </GroupItem>

            <ButtonItem
              horizontalAlignment="left"
              buttonOptions={{ text: 'Войти', type: 'success', useSubmitBehavior: true, disabled: formFetch }}
            />
          </Form>
        </form>
      </Item>
    </Box>
  );
};

export default observer(Login);
