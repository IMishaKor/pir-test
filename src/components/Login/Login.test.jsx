import React from 'react';
import { create, act } from 'react-test-renderer';

import Login from './Login';

jest.mock('../../../node_modules/devextreme-react/box');
jest.mock('../../../node_modules/devextreme-react/form');

describe('<Login/> ', () => {
  test('render', async () => {
    let component;
    await act(async () => {
      component = create(<Login />);
    });

    let tree = component.toJSON();
    console.log(component.getInstance());
    expect(tree).toMatchSnapshot();

    // await act(async () => {
    //   render(<Login />);
    // });
    // userEvent.type(screen.getByRole('textbox', { name: 'Password' }), 'couac');
    // let el = document.getElementById('div');
    // let sb = Form.getInstance(el);
    // console.log(el.innerHTML);
    // const { container } = render(<Login />);
    // console.log(container.getInstance());
    // const { container } = render(<Login />);
    // expect(container.innerHTML).toBeDefined();
    // expect(container.firstChild.classList.contains('header__items')).toBe(true);
  });
});
