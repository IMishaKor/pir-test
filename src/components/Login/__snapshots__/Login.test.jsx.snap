// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/>  render 1`] = `null`;
